tools/list_plugins

```diff
@@ -0,0 +1,64 @@
+#! /usr/bin/env python
+from itertools import chain
+import os.path
+import sys
+import xml.etree.ElementTree as ET
+
+import roslib
+import rospkg
+
+INDENT = "  "
+
+
+def list_plugins():
+    rospack = rospkg.RosPack()
+    to_check = rospack.get_depends_on('ed', implicit=False)
+    to_check.append("ed")  # also check for plugins in ed
+    first_file = True
+
+    for pkg in to_check:
+        m = rospack.get_manifest(pkg)
+        plugin_file = m.get_export('ed', 'plugin')
+        if not plugin_file:
+            continue
+        elif len(plugin_file) != 1:
+            print(f"Cannot load plugin [${pkg}]: invalid 'plugin' attribute", file=sys.stderr)
+            continue
+
+        plugin_file = plugin_file[0]
+        if not os.path.isfile(plugin_file):
+            print(f"Plugin file: '${plugin_file}' is not a file", file=sys.stderr)
+            continue
+        if not first_file:
+            print("")
+        print(f"{plugin_file}:")
+        first_file = False
```

Comment:
I would just add an  enter after each library. Then you would not need to keep track of whether it is the first file or not. 
If you do want to keep this formatting. Please separate the functionality which reads the files and the functionality for formatting as much as possible